### ISSUE TITLE
GUI or IDE not opening in Win11

### ISSUE THREAD

**Issue by Arjay-El** [VBAndCs/sVB-Small-Visual-Basic#4]
I am running Win11. Net framework is installed.

When I double click the .exe, I see the splashcreeen for a minute and then it closes, no gui/ide/form opens. Am I missing something?

PS: Thanks for keeping BASIC alive, Its still a worthy system.

**Comment by VBAndCs** [comment]
@Arjay-El 
Thanks for reporting this. I've heard about this before, but unfortunately I couldn't install Win11 on any of my PCs and laptops because they don't have the minimum hardware required to run win 11, and I tried all tips and tricks mentioned to bypass the win11 checks, but it still complains about the old processors I am using!
So, I need help here from any developer that has VS.NET installed on Win11 to [download the source code from here](https://github.com/VBAndCs/sVB-Small-Visual-Basic/archive/refs/heads/master.zip) and run it to find out if it has errors on win 11, or it will work fine and we just need to rebuild the exe for win11. I suspect that some component doesn't work well on Win11, but I need more info on that.
If you want to help on that, I can walk you through installing VS.NET (the Community Edition is free), and how to build the sVB source code.
Thanks.

**Comment by VBAndCs** [comment]
@Arjay-El 
As a workaround, you can activate the Hyper-V service on win 11. It is a virtual machine that allows you to create a virtual HDD and install any other operating system on, so you can install a lite 32-bit win 7, and run it from win 11 to run sVB on.

**Comment by Arjay-El** [comment]
Will give that a try. 

Also, I am not a programmer by trade but if you write instructions on how to recompile under Win11, I can give that a try. The majority of my compiler experience has been with fortran and matlab code for engineering analysis.

**Comment by VBAndCs** [comment]
@Arjay-El 
You can download VS .NET 2022 community for free from:
https://visualstudio.microsoft.com/vs/community/
While installing it, you only need to select the Desktop workload (you can modify the installation later to add more workloads if you need them).
Then download sVB source code from Github:
https://github.com/VBAndCs/sVB-Small-Visual-Basic/archive/refs/heads/master.zip
Unzip the file, then open the sVB.sln file with VS .NET and hit F5 to run the code. If the sVB runs correctly, you can find the compiled files in SmallBasicIDE\bin\Debug folder in the source code folder. The ToolBox and lib folders will be missing, so, you can copy them from the SmallBasicIDE\SB.Lib folder and past them to the debug folder.
It is better of course to build the source for release, but lets see if this works first. If any errors happened, please tell me about.
Thanks.

**Comment by Arjay-El** [comment]
I have visual studio community version previously loaded so I loaded the .sln file and ran a build (tried the F5 key but nothing seemed to happen so I chose build from the menu (it is set to debug, not release). There are two listed targets I see, svBCompiler and svb.Ide. Attached is a text file of the build errors and warnings.

**Comment by VBAndCs** [comment]
@Arjay-El 
Thanks. Please install [.NET Framework 4.8 developer pack])https://dotnet.microsoft.com/en-us/download/dotnet-framework/thank-you/net48-developer-pack-offline-installer) and try again.

**Comment by VBAndCs** [comment]
@Arjay-El 
This is an sVB version built on win-11. Try it and tell me if it works fine on your PC.
[sVB_Win11.zip](https://github.com/VBAndCs/sVB-Small-Visual-Basic/files/10416293/sVB_Win11.zip)

**Comment by Arjay-El** [comment]
Thanks for the update. Same behavior as the previous. I have not had time to go back and make another build attempt but I will try this weekend. I will delete VS completely and re-install the specific version number. Updates and extra extensions to an existing installation like I tried typically don’t seem to work as well as clean installs. I find Win11 has various compatibility problems that did not exist in Win10 (I use a lot of number crunching packages and its an endless cycle of workarounds). Unfortunately MS pushes users  into the new system. Maybe Win12 will correct it all.

 

Thanks.

**Comment by VBAndCs** [comment]
@Arjay-El 
I think your anti-virus or fire wall may be responsible for this issue. This version is built using VS on win 11 and works fine even on win 32, so, I see no issue in VS nor sVB. You better check your win 11 settings.

**Comment by VBAndCs** [comment]
@Arjay-El 
I want to try something. Remove the Lib and ToolBar folders from the sVB\bin folder then run sVB. I want to make sure that they are not the source of the issue.

**Comment by progamateur** [comment]
@Arjay-El
Hello,
Same problem with windows 10 22H2.
I solved it by modifying the sVB.exe.config file by adding a `<runtime>` section after the `<system.diagnostics>` section i.e. :

`<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <system.diagnostics>
        <sources>
            
            <source name="DefaultSource" switchName="DefaultSwitch">
                <listeners>
                    <add name="FileLog"/>
                    
                    
                </listeners>
            </source>
        </sources>
        <switches>
            <add name="DefaultSwitch" value="Information"/>
        </switches>
        <sharedListeners>
            <add name="FileLog" type="Microsoft.VisualBasic.Logging.FileLogTraceListener, Microsoft.VisualBasic, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL" initializeData="FileLogWriter"/>
            
            
        </sharedListeners>
    </system.diagnostics>
	<runtime>  
		<loadFromRemoteSources enabled="true"/>  
	</runtime>	
<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.8"/></startup></configuration>
`

I hope this will work for windows 11 as well...

**Comment by Arjay-El** [comment]
It seems to be working in Win11. Can you tell me why this fix worked? Thanks for your help, much appreciated to get this working!

**Comment by VBAndCs** [comment]
@Arjay-El @progamateur 
Thanks both for helping to fix this. I updated the sVB and the compiler projects to add this fix. You can download the Updated version from:
https://github.com/VBAndCs/sVB-Small-Visual-Basic/releases/download/v2.7/sVB.2.7.6.zip

This [article](https://learn.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/loadfromremotesources-element) seems to explain the issue. sVB uses Assembly.LoadFile to load external libraries from the sVB\bin\lib folder. This was trusted by default on .NET Framework 4, but Win11 seems to have a security update that made it untrusted!

**Comment by progamateur** [comment]
In the \bin directory, with the original sVB.exe.config file, it seems that an error is preventing sVB.exe from starting.
I was able to see the corresponding error message by typing the following command:
             sVB.exe 2>&1 | more.
This message gives information about the type of error (probably a FileLoadException) and suggests to consult [this link](http://go.microsoft.com/fwlink/?LinkId=155569) 
which miraculously gives the solution in the "Example" paragraph:
   The following example shows how to grant full trust to applications from remote sources.
`<configuration>
   <runtime>
      <loadFromRemoteSources enabled="true"/>
   </runtime>
</configuration>`

Hence the change to the sVB.exe.config file that I applied.

P.S I'm not a "pro" programmer.
